### DISCUSSION TITLE
What is the recommended way to change data via ItemsService within a cron-defined hook

### DISCUSSION THREAD

**Discussion by guidoka** [directus/directus#8182] [Archive]
in a hook-function with cron job i do not have schema or accountability to use ItemsService?

```
module.exports = function registerHook(context) {
    return {
		'cron(*/10 * * * * *)': async function () {
                const myItemService = new ItemsService('collection', { 
                     schema: ??, 
                     accountability: ?? 
                });
        },
    };

};
```
I have to say i'm new on directus, so if this solution is obvious, don't be so hard to me :-=)

**Comment by azrikahar** [comment]
The event handler function does have a context (not to be confused with the registerHook's context) as well, so you can do

```js
'cron(*/10 * * * * *)': async function ({ accountability, schema }) {
    const myItemService = new ItemsService('collection', { accountability, schema });
}
```

Link to docs for reference: https://docs.directus.io/guides/api-hooks/#event-handler-function

**Comment by guidoka** [comment]
thank you very much for your help.

but when i console.log(accountability, schema) i got both undefined


when i try
'cron(*/10 * * * * *)': async function ( params )

and console.log(params) a timestamp is in the output 
2021-09-21T03:22:30.942Z

**Reply by azrikahar** [reply]
It's my mistake, cron does behave differently. To get schema for cron hooks, you'll need to use the `getSchema()` in the registerHook's context, make sure to await it, like so:

_Note: feel free to do `registerHook({ getSchema, services })` instead_

```js
module.exports = function registerHook(context) {
	return {
		'cron(*/10 * * * * *)': async function () {
		        const schema = await context.getSchema();
			const myItemService = new context.services.ItemsService('collection', { schema });
		},
	};
};
```

Also note that since cron jobs are technically scheduled tasks on the server, we don't really have access to accountability (it's the server doing it, not someone/user doing it). If your concern is not having enough permission, not adding accountability just means it'll have full permissions to do what you want so don't worry about it.

**Reply by guidoka** [reply]
Thank you SO much - i didn't understand that i can call getSchema() and that accountability could be omitted :-) Great! Thank you!

**Reply by guidoka** [reply]
@azrikahar last answer is the solution (i cant mark it, only the thread)

**Reply by benhaynes** [reply]
CC @alesvaupotic for docs on this 📄

**Comment by lpkobamn** [comment]
can't find this in docs, can you help please?
